README.md

```diff
@@ -90,24 +90,87 @@ schema: tr.Schema = tr.data.tabular_sequence_testing_data.schema
 
 max_sequence_length, d_model = 20, 64
 
-# Define the input module to process the tabular input-features.
+# Define the input module to process the tabular input features.
 input_module = tr.TabularSequenceFeatures.from_schema(
     schema,
     max_sequence_length=max_sequence_length,
     continuous_projection=d_model,
     aggregation="concat",
     masking="causal",
 )
-# Define one or more prediction-tasks.
-prediction_tasks = tr.NextItemPredictionTask()
 
 # Define a transformer-config like the XLNet architecture.
 transformer_config = tr.XLNetConfig.build(
     d_model=d_model, n_head=4, n_layer=2, total_seq_length=max_sequence_length
 )
-model: tr.Model = transformer_config.to_torch_model(input_module, prediction_tasks)
+
+# Define the model block including: inputs, masking, projection and transformer block.
+body = tr.SequentialBlock(
+    input_module, 
+    tr.MLPBlock([d_model]), 
+    tr.TransformerBlock(transformer_config, masking=input_module.masking)
+)
+
+# Define the evaluation top-N metrics and the cut-offs
+metrics = [NDCGAt(top_ks=[20, 40], labels_onehot=True),  
+           RecallAt(top_ks=[20, 40], labels_onehot=True)]
+
+# Define a head with NextItemPredictionTask 
+head = tr.Head(
+    body,
+    tr.NextItemPredictionTask(weight_tying=True, metrics=metrics),
+    inputs=input_module,
+)
+
+# Get the end-to-end Model class 
+model = tr.Model(head)
+```
+
+The following code sample shows how to define and train an `XLNet` model for binary classification prediction task:
+
+```python
+from transformers4rec import torch as tr
+
+# Create a schema or read one from disk: tr.Schema().from_json(SCHEMA_PATH).
+schema: tr.Schema = tr.data.tabular_sequence_testing_data.schema
+
+max_sequence_length, d_model = 20, 64
+
+# Define the input module to process the tabular input features.
+input_module = tr.TabularSequenceFeatures.from_schema(
+    schema,
+    max_sequence_length=max_sequence_length,
+    continuous_projection=d_model,
+    aggregation="concat",
+    masking="causal",
+)
+
+# Define XLNetConfig class and set default parameters for HF XLNet config  
+transformer_config = tr.XLNetConfig.build(
+    d_model=d_model, n_head=4, n_layer=2, total_seq_length=max_sequence_length
+)
+# Define the model block including: inputs, masking, projection and transformer block.
+body = tr.SequentialBlock(
+    input_module, tr.MLPBlock([64]), tr.TransformerBlock(transformer_config, masking=input_module.masking)
+)
+
+# Define a head with BinaryClassificationTask
+head = tr.Head(
```

Comment:
My $0.02 is that the difference from the preceding example is too small to justify putting it here.

I've started updating the docstring for the BinaryClassificationTask to show this code sample.

Can you tell me what `tm` in the `metrics` represents?

Reply:
I see it now, it's `import torchmetrics as tm`.

Reply:
`import torchmetrics as tm` yes. should I add it that line?

Reply:
@mikemckiernan `I've started updating the docstring for the BinaryClassificationTask to show this code sample.` does that mean you have a PR as well, therefore we dont need this PR?  Or where do you think we should add it this code snippet?